### PR TITLE
refactor(editor): integrate findreplace and spellcheck

### PR DESCRIPTION
**Pull request by petrjasek**
instead of kind of stand alone those functions are now integrated
in the editor, so you can switch between those.
you can trigger those via editor service on all registered scopes.

instead of using input event it's now using keydown/up to detect
changes on items which can be only triggered by user.

SD-2793

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/editor/find-replace.js [hunk 1]
```diff
@@ -22,32 +22,36 @@ function FindReplaceDirective($timeout, $rootScope, editor) {
...
             scope.replace = function() {
-                editor.command.replace(scope.to || '');
+                editor.replace(scope.to || '');
+                editor.selectNext();
             };
 
             scope.replaceAll = function() {
-                editor.command.replaceAll(scope.to || '');
+                editor.replaceAll(scope.to || '');
             };
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/editor/find-replace.js]: I suggest adding docstrings to the modified methods.

File: client/app/scripts/superdesk/editor/editor.js [hunk 2]
```diff
@@ -11,318 +11,485 @@
...
+ * @param {Node} node
+ * @param {numeric} offset
+ * @return {Object} {node: {Node}, offset: {numeric}}
+ */
+function findTextNode(node, offset) {
+    var tree = document.createTreeWalker(node, NodeFilter.SHOW_TEXT);
+    var currentLength;
+    var currentOffset = 0;
+    while (tree.nextNode()) {
+        tree.currentNode.textContent = tree.currentNode.textContent.replace(String.fromCharCode(65279), '');
```
Comment (plamut) [inline client/app/scripts/superdesk/editor/editor.js]: It's not clear what character `String.fromCharCode(65279)` represents, I suggest creating an alias for it.

File: client/app/scripts/superdesk/editor/editor.js [hunk 3]
```diff
@@ -400,48 +567,76 @@ angular.module('superdesk.editor', [])
...
+                editor.registerScope(scope);
+
+                var TYPING_CLASS = 'typing';
+
+                var ARROWS = {
+                    37: 1,
+                    38: 1,
+                    39: 1,
+                    40: 1
+                };
```
Comment (plamut) [inline client/app/scripts/superdesk/editor/editor.js]: Perhaps it makes sense to freeze the object?

**Comment by plamut** [comment]
Looks more or less good. :+1:

File: client/app/scripts/superdesk/editor/editor.js (hunk 2 above)
Comment (petrjasek) [inline client/app/scripts/superdesk/editor/editor.js]: it's not clear to me neither. anyone has an idea?

Comment (amagdas) [inline client/app/scripts/superdesk/editor/editor.js]: http://stackoverflow.com/questions/6538203/how-to-avoid-echoing-character-65279-in-php-this-question-also-relates-to-java

Comment (plamut) [inline client/app/scripts/superdesk/editor/editor.js]: I didn't know without googling for it, hence the remark. :)

**Comment by amagdas** [comment]
:100: :trophy: :heart: :blue_heart: :yellow_heart: :green_heart:

**Pull request merged**